lib/auth/trustedcluster.go

```diff
@@ -456,3 +442,64 @@ func (v *ValidateTrustedClusterResponseRaw) ToNative() (*ValidateTrustedClusterR
 		CAs: cas,
 	}, nil
 }
+
+// activateCertAuthority will activate both the user and host certificate
+// authority given in the services.TrustedCluster resource.
+func (a *AuthServer) activateCertAuthority(t services.TrustedCluster) error {
+	err := a.ActivateCertAuthority(services.CertAuthID{Type: services.UserCA, DomainName: t.GetName()})
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	return trace.Wrap(a.ActivateCertAuthority(services.CertAuthID{Type: services.HostCA, DomainName: t.GetName()}))
+}
+
+// deactivateCertAuthority will deactivate both the user and host certificate
+// authority given in the services.TrustedCluster resource.
+func (a *AuthServer) deactivateCertAuthority(t services.TrustedCluster) error {
+	err := a.DeactivateCertAuthority(services.CertAuthID{Type: services.UserCA, DomainName: t.GetName()})
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	return trace.Wrap(a.DeactivateCertAuthority(services.CertAuthID{Type: services.HostCA, DomainName: t.GetName()}))
+}
+
+// createReverseTunnel will create a services.ReverseTunnel givenin the
+// services.TrustedCluster resource.
+func (a *AuthServer) createReverseTunnel(t services.TrustedCluster) error {
+	reverseTunnel := services.NewReverseTunnel(
+		t.GetName(),
+		[]string{t.GetReverseTunnelAddress()},
+	)
+	return trace.Wrap(a.UpsertReverseTunnel(reverseTunnel))
+}
+
+// canChangeState checks if the state change is allowed or not. If not, returns
+// an error explaining the reason.
+func canChangeState(ec services.TrustedCluster, nc services.TrustedCluster) error {
```

Comment:
I would make this a method of Trusted cluster, `TrustedCluster.CanChangeStateTo(other) -> error`